File: codeforces/357.2/B.js
;(function () {
	var n = 1 * readline()
	for (var a = 0; a <= n; a += 1234567)
		for (var b = 0; a + b <= n; b += 123456)
			if ((n - a - b) % 1234 === 0)
				return print('YES');
	print('NO');
})();

File: codeforces/357.2/B.py
n = int(input())
a = 0
while a <= n:
	b = 0
	while a + b <= n:
		if (n - a - b) % 1234 == 0:
			print('YES')
			exit(0)
		b += 123456
	a += 1234567
print('NO')
